### PR TITLE
test row deleted in devBranch shoudl be deleted in default after merge

### PR DESCRIPTION
**Pull request by romanbracinik**
Jira: CT-XXX
KBC:  https://github.com/keboola/connection/pull/5049

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

### CODE REVIEW

**Comment by romanbracinik** [comment]
Davam na review pada to na inej veci ktoru snad v priebehu fixneme

File: tests/Backend/SOX/MergeRequestsTest.php [hunk 1]
```diff
@@ -682,6 +682,106 @@ public function testMrWithConflictCantBeMergedButAfterResetCan(): void
         $this->assertBranchIsDeleted($branchId);
     }
 
+    public function testDeletedRowInBranchIsDeletedInDefaultAfterMerge(): void
```
Comment (jirkasemmler) [inline tests/Backend/SOX/MergeRequestsTest.php:685]: @romanbracinik frci ti to ? me to pada na `Row firstRow already exists`. nema ten `rowId` byt nejak unique?

**Review by jirkasemmler — CHANGES_REQUESTED** [review]

File: tests/Backend/SOX/MergeRequestsTest.php (hunk 1 above)
Comment (romanbracinik) [reply tests/Backend/SOX/MergeRequestsTest.php:685]: Row ma unique na `idProjec,IdBranch,IdComponent,configurationId,rowId`

Screenshot 2024-05-13 at 15 56 55

Teraz som to znova spustil a preslo to

Comment (jirkasemmler) [reply tests/Backend/SOX/MergeRequestsTest.php:685]: tak su asi levej. jdu si to debugnout

**Review by jirkasemmler — APPROVED** [review]
nefrci to mi to, ale kldyz davas screenshot, tak za me ok. chyba bude u me. tagni to prosim

**Comment by romanbracinik** [comment]
@jirkasemmler https://dev.azure.com/keboola-dev/Connection%20build/_build/results?buildId=46019&view=logs&j=2e32e92a-b774-5b2a-1ae4-fd294511f88b&t=43215129-de12-593b-bf06-9d12e014da10 tag spadla jedna suita na inej veci.

**Comment by jirkasemmler** [comment]
formalni re-palecek ✅

**Comment by romanbracinik** [comment]
rebase

**Pull request merged**